### PR TITLE
Closes #1012: Fixes footer styling/templates

### LINKED ISSUE #1012: Styling on footer links broken
Looks like the changes in this PR https://github.com/az-digital/az_quickstart/pull/940 has broken something with the styling of the links/headers in the footer. 

## Problem/Motivation
Some problem with templates not being used (probably due to naming issues). 

### Describe the bug
image

Looks like issue with the template names
Making a separate issue for the white background on footer links (introduced by the work we did on putting nav over text on media)

### PR DESCRIPTION
**Pull request by smayden**
## Related Issue
#1012

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I've added this Pull Request to the AZ Quickstart Github project and set it to "Review in progress".
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: themes/custom/az_barrio/az_barrio.theme [hunk 1]
```diff
@@ -53,6 +54,34 @@ function az_barrio_preprocess_form_element(array &$variables) {
...
+ *  Footer menu links styles, using default theme templates.
+ */
+function az_barrio_preprocess_block(array &$variables) {
+  if (isset($variables['elements']['#base_plugin_id']) && $variables['elements']['#base_plugin_id'] === 'menu_block') {
+    if (!empty($variables['elements']['#id'])) {
+      $block = Block::load($variables['elements']['#id']);
+      if(!empty($block) && $block->getRegion() === 'footer') {
+        $variables['content']['#attributes']['class'][] = 'justify-content-end';
+      }
+      if(!empty($block) && $block->getRegion() === 'footer_sub') {
```
Comment (trackleft) [inline themes/custom/az_barrio/az_barrio.theme:70]: Could simplify with `if(!empty($block)` around both region checks.

File: themes/custom/az_barrio/config/optional/block.block.footerinformationfor.yml [hunk 2]
```diff
@@ -4,23 +4,31 @@ dependencies:
...
   module:
+    - block_class
     - menu_block
   theme:
     - az_barrio
-id: az_barrio_footer_menu_info
+third_party_settings:
+  block_class:
+    classes: 'col-12 col-sm-4 col-md-3'
+id: footerinformationfor
```
Comment (trackleft) [inline themes/custom/az_barrio/config/optional/block.block.footerinformationfor.yml:14]: change to `az_barrio_footer_menu_info`

**Comment by danahertzberg** [comment]
Dana and Shoshana will determine styles needed for nav items. Troy will apply them with a preprocess function for the block in that region.

**Comment by smayden** [comment]
@tadean @trackleft 
Any suggestions on how to get these items incorporated into this PR? 

- [ ] Sub footer area: Add d-flex px-0 to the li 
- [ ] In both footer and sub footer region - Set background area to transparent

**Comment by joshuasosa** [comment]
> * [ ]  Sub footer area: Add d-flex px-0 to the li

In az_barrio_preprocess_block, this worked in my local testing:
```
foreach ($variables['elements']['content']['#items'] as $key => $item) {
  $variables['elements']['content']['#items'][$key]['attributes']->addClass('d-flex', 'px-0');
}
```

> * [ ]  In both footer and sub footer region - Set background area to transparent

Are you wanting to set link backgrounds to transparent? You can use the utility `.bg-transparent`

**Comment by tadean** [comment]
@joshuasosa the following seemed to work well, thanks!
```
foreach ($variables['elements']['content']['#items'] as $key => $item) {
  $variables['elements']['content']['#items'][$key]['attributes']->addClass('d-flex', 'px-0');
}
```

File: themes/custom/az_barrio/az_barrio.theme [hunk 3]
```diff
@@ -522,12 +523,81 @@ function az_barrio_theme_suggestions_menu_alter(array &$suggestions, array $vari
...
     }
   }
+
+  if (isset($variables['elements']['#base_plugin_id']) && $variables['elements']['#base_plugin_id'] === 'menu_block') {
+    if (!empty($variables['elements']['#id'])) {
+      $block = Block::load($variables['elements']['#id']);
+      if (!empty($block)) {
+        $region = $block->getRegion();
+        // Default classes for footer region.
+        if($region === 'footer') {
```
Comment (joeparsons) [inline themes/custom/az_barrio/az_barrio.theme:546]: Nitpick:
```suggestion
        if ($region === 'footer') {
```

File: themes/custom/az_barrio/az_barrio.theme [hunk 4]
```diff
@@ -522,12 +523,81 @@ function az_barrio_theme_suggestions_menu_alter(array &$suggestions, array $vari
...
+          $variables['content']['#attributes']['class'][] = 'nav-utility';
+          $variables['content']['#attributes']['class'][] = 'd-flex';
+          $variables['content']['#attributes']['class'][] = 'justify-content-end';
+          $variables['title_attributes']['class'][] = 'h6';
+          $variables['title_attributes']['class'][] = 'mt-md-3';
+          $variables['title_attributes']['class'][] = 'text-uppercase';
+        }
+
+        // Default classes for footer_sub region.
+        if($region === 'footer_sub') {
```
Comment (joeparsons) [inline themes/custom/az_barrio/az_barrio.theme:571]: Nitpick:
```suggestion
        if ($region === 'footer_sub') {
```

File: themes/custom/az_barrio/config/optional/block.block.az_barrio_footer_menu_resources.yml [hunk 5]
```diff
@@ -4,23 +4,31 @@ dependencies:
   config:
     - system.menu.az-footer-resources
   module:
+    - block_class
     - menu_block
   theme:
     - az_barrio
+third_party_settings:
+  block_class:
+    classes: 'col-6 col-sm-4 col-md-2'
```
Comment (danahertzberg) [inline themes/custom/az_barrio/config/optional/block.block.az_barrio_footer_menu_resources.yml:13]:
```suggestion
    classes: 'col-12 col-sm-6 col-md-2'
```

File: themes/custom/az_barrio/config/optional/block.block.az_barrio_footer_menu_topics.yml [hunk 6]
```diff
@@ -4,23 +4,31 @@ dependencies:
   config:
     - system.menu.az-footer-topics
   module:
+    - block_class
     - menu_block
   theme:
     - az_barrio
+third_party_settings:
+  block_class:
+    classes: 'col-12 col-sm-8 col-md-5'
```
Comment (danahertzberg) [inline themes/custom/az_barrio/config/optional/block.block.az_barrio_footer_menu_topics.yml:13]:
```suggestion
    classes: 'col-12 col-sm-6 col-md-5'
```

File: themes/custom/az_barrio/config/optional/block.block.az_barrio_footer_menu_info.yml [hunk 7]
```diff
@@ -4,23 +4,31 @@ dependencies:
   config:
     - system.menu.az-footer-information-for
   module:
+    - block_class
     - menu_block
   theme:
     - az_barrio
+third_party_settings:
+  block_class:
+    classes: 'col-12 col-sm-4 col-md-3'
```
Comment (danahertzberg) [inline themes/custom/az_barrio/config/optional/block.block.az_barrio_footer_menu_info.yml:13]:
```suggestion
    classes: 'col-12 col-sm-6 col-md-3'
```

**Review by danahertzberg — CHANGES_REQUESTED** [review]

**Review by danahertzberg — APPROVED** [review]
Muy bien!

**Review by trackleft — APPROVED** [review]

**Pull request merged**